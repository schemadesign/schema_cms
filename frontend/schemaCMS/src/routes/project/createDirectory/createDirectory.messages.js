/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'project.createDirectory.title',
    defaultMessage: 'Directories',
  },
  subTitle: {
    id: 'project.createDirectory.subTitle',
    defaultMessage: 'Create',
  },
  directoryFieldName: {
    id: 'project.createDirectory.directoryFieldName',
    defaultMessage: 'Directory Name',
  },
  directoryFieldPlaceholder: {
    id: 'project.createDirectory.directoryFieldPlaceholder',
    defaultMessage: 'Name',
  },
  createDirectory: {
    id: 'project.createDirectory.createDirectory',
    defaultMessage: 'Create',
  },
  cancel: {
    id: 'project.createDirectory.cancel',
    defaultMessage: 'Cancel',
  },
});
