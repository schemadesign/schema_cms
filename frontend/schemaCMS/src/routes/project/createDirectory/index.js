export { default as CreateDirectory } from './createDirectory.container';
