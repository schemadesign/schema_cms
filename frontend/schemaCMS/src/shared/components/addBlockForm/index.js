export { AddBlockForm } from './addBlockForm.component';
