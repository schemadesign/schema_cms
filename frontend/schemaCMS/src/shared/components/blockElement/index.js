export { BlockElement } from './blockElement.component';
