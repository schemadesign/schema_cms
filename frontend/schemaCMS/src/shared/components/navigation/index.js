export { NavigationContainer, BackArrowButton, PlusButton, BackButton, NextButton } from './navigation.component';
