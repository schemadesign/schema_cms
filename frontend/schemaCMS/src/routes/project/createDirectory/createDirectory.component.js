import React, { Fragment, PureComponent } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { path } from 'ramda';

import { Form } from './createDirectory.styles';
import messages from './createDirectory.messages';
import { TextInput } from '../../../shared/components/form/inputs/textInput';
import { DIRECTORY_NAME } from '../../../modules/directory/directory.constants';
import { TopHeader } from '../../../shared/components/topHeader';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { BackButton, NavigationContainer, NextButton } from '../../../shared/components/navigation';

export class CreateDirectory extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    values: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleBlur: PropTypes.func.isRequired,
  };

  getProjectId = () => path(['match', 'params', 'projectId'], this.props);

  handleBackClick = () => this.props.history.push(`/project/${this.getProjectId()}/directory`);

  render() {
    const { intl, values, handleSubmit, handleChange, ...restProps } = this.props;
    const headerTitle = <FormattedMessage {...messages.title} />;
    const headerSubtitle = <FormattedMessage {...messages.subTitle} />;

    return (
      <Fragment>
        <TopHeader headerTitle={headerTitle} headerSubtitle={headerSubtitle} />
        <ContextHeader title={headerTitle} subtitle={headerSubtitle} />
        <Form onSubmit={handleSubmit}>
          <TextInput
            value={values[DIRECTORY_NAME]}
            onChange={handleChange}
            name={DIRECTORY_NAME}
            label={intl.formatMessage(messages.directoryFieldName)}
            placeholder={intl.formatMessage(messages.directoryFieldPlaceholder)}
            fullWidth
            isEdit
            {...restProps}
          />
          <NavigationContainer>
            <BackButton id="backBtn" onClick={this.handleBackClick}>
              <FormattedMessage {...messages.cancel} />
            </BackButton>
            <NextButton id="createDirectoryBtn" type="submit" disabled={!restProps.isValid}>
              <FormattedMessage {...messages.createDirectory} />
            </NextButton>
          </NavigationContainer>
        </Form>
      </Fragment>
    );
  }
}
