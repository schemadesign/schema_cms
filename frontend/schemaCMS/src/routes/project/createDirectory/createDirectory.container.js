import { connect } from 'react-redux';
import { promisifyRoutine, bindPromiseCreators } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose, path } from 'ramda';
import { injectIntl } from 'react-intl';
import { withFormik } from 'formik';

import { CreateDirectory } from './createDirectory.component';
import { DirectoryRoutines, selectDirectoryName } from '../../../modules/directory';
import {
  DIRECTORY_FORM,
  DIRECTORY_NAME,
  DIRECTORY_SCHEMA,
  INITIAL_VALUES,
} from '../../../modules/directory/directory.constants';
import { errorMessageParser } from '../../../shared/utils/helpers';

const mapStateToProps = createStructuredSelector({
  directoryName: selectDirectoryName,
});

export const mapDispatchToProps = dispatch =>
  bindPromiseCreators(
    {
      createDirectory: promisifyRoutine(DirectoryRoutines.create),
    },
    dispatch
  );

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter,
  injectIntl,
  withFormik({
    displayName: DIRECTORY_FORM,
    isInitialValid: true,
    enableReinitialize: true,
    mapPropsToValues: ({ directoryName }) => ({
      ...INITIAL_VALUES,
      [DIRECTORY_NAME]: directoryName,
    }),
    validationSchema: () => DIRECTORY_SCHEMA,
    handleSubmit: async (data, { props, setSubmitting, setErrors }) => {
      try {
        setSubmitting(true);
        const projectId = path(['match', 'params', 'projectId'], props);

        await props.createDirectory({ projectId, ...data });
      } catch (errors) {
        const { formatMessage } = props.intl;
        const errorMessages = errorMessageParser({ errors, messages, formatMessage });

        setErrors(errorMessages);
      } finally {
        setSubmitting(false);
      }
    },
  })
)(CreateDirectory);
