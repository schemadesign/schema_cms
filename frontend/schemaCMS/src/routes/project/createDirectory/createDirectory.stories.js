import React from 'react';
import { storiesOf } from '@storybook/react';

import { CreateDirectory } from './createDirectory.component';

const defaultProps = {};

storiesOf('CreateDirectory', module).add('Default', () => <CreateDirectory {...defaultProps} />);
