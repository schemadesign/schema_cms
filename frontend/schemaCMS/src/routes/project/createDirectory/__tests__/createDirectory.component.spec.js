import React from 'react';
import { shallow } from 'enzyme';

import { CreateDirectory } from '../createDirectory.component';
import { defaultProps } from '../createDirectory.stories';
import { BackButton } from '../../../../shared/components/navigation';
import { Form } from '../createDirectory.styles';

describe('CreateDirectory: Component', () => {
  const component = props => <CreateDirectory {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should return to directory list', () => {
    jest.spyOn(defaultProps.history, 'push');
    const wrapper = render();
    wrapper.find(BackButton).simulate('click');

    expect(defaultProps.history.push).toHaveBeenCalledWith('/project/1/directory');
  });

  it('should submit form', () => {
    jest.spyOn(defaultProps, 'handleSubmit');
    const wrapper = render();
    wrapper.find(Form).simulate('submit');

    expect(defaultProps.handleSubmit).toHaveBeenCalled();
  });
});
