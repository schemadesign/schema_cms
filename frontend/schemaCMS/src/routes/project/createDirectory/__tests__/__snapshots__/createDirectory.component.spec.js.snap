// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDirectory: Component should render correctly 1`] = `
<Fragment>
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Create"
        id="project.createDirectory.subTitle"
        values={Object {}}
      />
    }
    headerTitle={
      <FormattedMessage
        defaultMessage="Directories"
        id="project.createDirectory.title"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Create"
        id="project.createDirectory.subTitle"
        values={Object {}}
      />
    }
    title={
      <FormattedMessage
        defaultMessage="Directories"
        id="project.createDirectory.title"
        values={Object {}}
      />
    }
  />
  <styled.form
    onSubmit={[Function]}
  >
    <TextInput
      checkOnlyErrors={false}
      errors={Object {}}
      fullWidth={true}
      handleBlur={[Function]}
      history={
        Object {
          "push": [Function],
        }
      }
      isEdit={true}
      label="Directory Name"
      match={
        Object {
          "params": Object {
            "projectId": "1",
          },
        }
      }
      name="name"
      onChange={[Function]}
      placeholder="Name"
      touched={Object {}}
    />
    <NavigationContainer
      hideOnDesktop={false}
      right={false}
    >
      <BackButton
        disabled={false}
        id="backBtn"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Cancel"
          id="project.createDirectory.cancel"
          values={Object {}}
        />
      </BackButton>
      <NextButton
        disabled={true}
        id="createDirectoryBtn"
        loading={false}
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Create"
          id="project.createDirectory.createDirectory"
          values={Object {}}
        />
      </NextButton>
    </NavigationContainer>
  </styled.form>
</Fragment>
`;
